\documentclass{article}
\usepackage{natbib}
\usepackage{subfiles}
\usepackage{booktabs}
\usepackage{multirow}
\usepackage{makecell}
\usepackage{graphicx}
\graphicspath{{../Figures/}}

\bibpunct{(}{)}{,}{a}{}{;}

\title{Linking microbial communities to ecosystem functions: what we can learn
from genotype-phenotype mapping in organisms} 

\author{Andrew H. Morris, Brendan J. M. Bohannan}

\begin{document}

\maketitle



Deadline: August 1st

Theme: Conceptual Challenges in Microbial Community Ecology

Journal: Philosophical Transactions B

Max length: 8-9 pages

Data must be submitted as supplementary information in a repository with
accession number, link, or DOI

Supplement: They have encouraged us to include additional information including
data, tables, figures, text, images, videos, or recordings.

\section*{Abstract}



\section{Introduction} 

Microorganisms mediate many biogeochemical processes and yet the link between
microbial biodiversity and ecosystem function remains unknown. 
Ecologists have investigated the effects of changing
biodiversity on ecosystem function for decades, but this research has primarily
focused on linking terrestrial plant community richness with productivity
\citep{hooper2005}. More recently, there has been an interest in linking
microbial community structure to the functioning of ecosystems both inside and
outside of a host, such as disease states, plant productivity,
and soil greenhouse gas emissions \citep{schimel1998, singh2010}. 
However, it remains unclear to what extent 
microbial community composition is important for determining the rates of ecosystem-scale
functions.  This uncertainty limits our ability to predict and manage host
health and global biogeochemical cycles. To
address this challenge, we propose integrating traditional
biodiversity-ecosystem function research with ideas from genotype-phenotype
mapping in population genetics. This connection will help determine
whether microbial community
composition alters ecosystem-scale function independent
of the underlying environmental variation.

If local selection always optimizes the available biodiversity to maximize
ecosystem function, then microbial community composition should not matter for
predicting the rates of microbially-mediated functions. Instead, 
the rates of these functions should be determined by the
underlying environmental variation. In this case, the microbial community simply
acts as a conduit through which the abiotic
environment alters ecosystem function. However, experiments that manipulate the
connection between environmental factors and microbial community composition
such as environmental treatments, common gardens, and reciprocal transplants
have shown different rates of ecosystem functions for different microbial
communities under the same conditions \citep{reed2007}. 
This has been observed for plant decomposition
\citep{strickland2009}, plant phenology \citep{panke-buisse2015}, soil nitrogen
cycling \citep{balser2005}, and soil greenhouse gas emissions
\citep{cavigelli2000}. Therefore, variation in microbial community composition
seems to generate variation in ecosystem function. However, in order to
incorporate microorganisms into ecosystem models, 
we should distinguish between the portion
of microbial biodiversity that is contributing to ecosystem function as a
consequence of the environmental conditions and that contributing uniquely due to
variation in the community independent of the environment. 

Most studies of microbial community function focus on one of two aspects of
microbial community structure that are hypothesized to predict
ecosystem function. The first aspect is functional gene or transcript abundance.
In this case, qPCR or shotgun metagonemic sequencing is used to estimate the
abundance of a gene
or transcript that is a putative marker for a microbial process. For example,
the gene \textit{mcrA}, which encodes the enzyme that performs the final step in
methanogenesis, is hypothesized to predict the rate of
methane production. Other examples include \textit{pmoA} and methanotrophy,
\textit{nifH} and nitrification, and \textit{nosZ} and denitrification. The
rationale for this hypothesis is that the abundance of a functional group (as
inferred from marker gene abundance) should correspond to the rate of the
process performed by that group. There are examples where this relationship
holds \citep{freitag2009, freitag2010, schnyder2018}. However, a review of these
studies found that the abundance of a functional gene or transcript is rarely
correlated with the rate of the corresponding process, with most effects either
negative or not significant \citep{rocca2015}.

The second aspect of microbial community structure hypothesized to predict
ecosystem function is taxonomic or functional
diversity. Diversity is either estimated from sequence variants of a barcode
gene such as the 16S rRNA gene or manipulated through some proxy
of diversity such as sequential dilution or varying filter sizes. Studies that
experimentally manipulate diversity generally find a relationship between
the applied diversity treatment and ecosystem functions including methanotrophy
\citep{schnyder2018}, phosphorus leaching, greenhouse gas emissions
\citep{wagg2014}, and decomposition \citep{maron2018}. However, in each of these
cases, richness is confounded with other factors such as presence or absence of
major phylogenetic groups (nematodes, fungi) or abundance of microbial cells. In
addition, assembled microbial communities pose the same challenges as
macroorganismal diversity experiments, for example lacking replication at the highest
diversity treatment such that that treatment will always have the most
productive taxon. A review of microbial diversity studies shows that
microbial community taxonomic and functional diversity add little variance
explained to models of ecosystem function. Overall, functional gene
abundance and community diversity improve models of ecosystem function less than
one third of the time and increase variance explained by an average of only 8 percentage
points \citep{graham2016}.

These approaches follow on the tradition of macroorganismal
biodiversity-ecosystem function research by measuring or manipulating diversity
and comparing it to function or by looking at the association between richness
or abundance of a functional group (e.g. plants) and the function performed by
that group (e.g. productivity). However, in the case of microbes, our knowledge
of functional groups is quite limited. We cannot be sure that we are accurately
estimating the abundance or diversity of soil microbial functional groups the
way we can plants in a grassland. In addition, we do not know the function of
most microbial taxa and metagenomic sequences recovered from environmental 
samples. Because of this, we should look more
agnostically at the microbial community to identify taxa or groups of taxa that
are important for predicting the rate of ecosystem function. We can accomplish
this by drawing on ideas from population genetics.

The challenge of mapping microbial community structure to ecosystem function is
analogous to the challenge of identifying the genetic basis of complex traits in
organismal populations. In the latter case, a population exhibits variation in a
phenotype (e.g. height or disease state) as well as variation in potentially
thousands of single nucleotide polymorphisms
(SNPs). To identify the genetic basis for a trait, investigators sample from
this population and correlate phenotype with genotype. In most cases, there are
many more loci than individuals and we do not know whether the SNPs are causally
linked or are simply in linkage disequilibrium with a causal mutation. 
Genetic loci identified in these genome-wide association
studies can be the target for future research to dissect the function of those
genes, can be added or removed to test their association with the trait, or can
be used as markers for that disease or phenotype even if it has no known
mechanistic association. This mirrors the problem of microbial
structure-function mapping. In this case, a population of ecosystems exhibit
variation in a particular trait or ecosystem function putatively mediated by
microorganisms as well as variation in metagenomic composition. Investigators
could sample from this population of ecosystems and correlate the rate of an
ecosystem function with microbial community composition or gene composition. 
In order to do this, microbial ecologists should incorporate similar
methodological and statistical approaches as population geneticists in order to
make robust associations between microbes and ecosystem function.

Population geneticists trying to make robust associations between genetic loci
and phenotype must consider several points when designing a genome-wide
association study. First is
to identify the precise trait expected to be influenced by genetic variation. 
Then, to choose a population that exhibits
substantial variation in the trait of interest. This population should also be
otherwise relatively homogeneous to avoid other factors that might influence the
trait. After sampling, statistical methods should take into account population
stratification due to shared ancestry or local selection that can lead to spurious
associations between genes and traits. For example, both human height and
genotype vary systematically across Europe in a latitudinal gradient, which can
lead to spurious genotype-phenotype associations \citep{novembre2008}. 

Positive associations between genes and
phenotypes can arise because of true associations, through linkage
disequilibrium with true genes, or through admixture \citep{lander1994}.
Microbial structure-function connections could pose these same problems. There
could be true associations between a microorganism and the ecosystem function
performed by that taxon, or there could be systematic
co-occurrence patterns between one taxon and a causal taxon,
or there could be admixture due to shared history or selection within
a subset of ecosystems. This is true for 16S amplicons and for
genes in metagenomic (or other -omic) datsets. 
Robust structure-function associations rely on
controlling for the underlying geographic and ecological covariance.

Organisms exhibit population differentiation due to geographic and ecological
processes \citep{wright1943}. Neutral processes such as local genetic drift and
reduced gene flow due to geographic separation can lead to population
differentiation. In addition, environmental distance can lead to selection
against migrants \citep{hendry2004} and local selection on endemic populations
to generate variation. Microorganisms also exhibit biogeographic patterns due to
ecological selection and geographic separation \citep{martiny2006}. If dispersal
is non-limiting then the most productive species should always be present and
composition should not matter for local ecosystem function. However, if important
functional groups are dispersal-limited, lost to drift, or are unable to
colonize a site, then there could be important functional consequences of
community differentiation due to geographic or ecological isolation.

There are a number of ways that environmental or spatial distance could obscure
microbial BEF relationships. For example, historical effects of geographic
isolation such as dispersal limitation could lead to spurious associations
between the abundance of taxa and the rate of ecosystem function. Community
stratification where taxa co-occur due to historical effects or local selection.
Environmental selection can also shape microbial community
structure. For example, pH is known to regulate both broad biodiversity patterns for
microbial communities \citep{fierer2006} and methane cycling dynamics
\citep{ye2012}. This could lead to microbial taxa that are correlated with
methane cycling, but are mechanistically unrelated. In fact, if the rate of
ecosystem function is high in one set of ecosystems, function will be
associated with any taxa that are abundant in those ecosystems. 
In order to understand the biological basis of ecosystem function, we need to
separate the close association between environmental variation, community
composition, and ecosystem function in order to identify what aspects of
microbial biodiversity determine the rate of ecosystem function.

Successfully connecting microbial community structure to ecosystem function
depends on being precise about the variation we want to explain. The first and
often overlooked step is choosing a
clearly defined function that exhibits considerable variation across ecosystems. 
Next, be intentional about choosing the
population to be sampled to maximize the amount of variation. For example, how
might ecosystems that are otherwise quite similar vary in function?
Alternatively, how might one ecosystem change in response to a single
environmental change? This could be achieved by sampling across similar ecosystems 
or by imposing experimental conditions that minimize environmental variation or
vary only abiotic factors of interest. 
It is important to take an agnostic approach towards identifying taxa that are
indicators of ecosystem function. It is also important to remember that any
relationships identified in this manner are provisional until further tests are
performed. The work is not done until the organism has been manipulated through
addition, knockout, or variation in abundance.

In this paper we propose integrating microbial biodiversity-ecosystem function
research with ideas and techniques from population genetics to advance our
understanding of the role of microbial commuities in the functioning of
ecosystems. Perhaps the most important part of that discussion is to define the
question carefully. In microbial ecology
structure-function research is focused on answering the question: what is the
unique contribution of the microbial community to ecosystem function? This is of
great interest for two reasons. First, because it addresses what is the singular
contribution of biodiversity to the functioning of ecosystems. Second, on a
practical level, because answering that question will be essential to
incorporate a microbial component into ecosystem models.

Is variation in microbial community
structure associated with differences in ecosystem function across space,
time, and environments?

\section{Methods}

\subsection{Sample collection}

\subsection{Determination of the rates of ecosystem functions}

\subsection{Measurement of environmental variables}

\subsection{Sequencing and bioinformatic processing}

DADA2 pipeline for amplicon sequence variant (ASV) assignment.

We used the variance stabilizing transformation from the \texttt{DESeq2}
package to adjust for differences in sequencing depth between samples. 

\subsection{Statistical analysis}

All statistical analyses were performed using the \texttt{R} statistical
environment. 
We first applied traditional tests of microbial biodiversity-ecosystem function
relationships. To do this, we fit ordinary least squares regression models with
the rate of ecosystem function as the response with one predictor for each
model. These predictors include functional gene and transcript abundance,
diversity measures including richness, Shannon's diversity index, and Simpson's
diversity index, measures of microbial community composition using the first
several principal component axes from a principal component
analysis (PCA) chosen based on visual inspection of a screeplot incpluding a PCA for
abundance data and a PCA for presence/absence data, and several 
environmental variables. 

Next, we used a statistical approach common in population genetics genome-wide
association studies. We fit variance component mixed models from the
\texttt{varComp} package to test whether variation in microbial
community composition expains a significant portion of the variation in
ecosystem function \citep{qu2013}. First, we fit an intercept-only model with
each random effect in all combinations. Variance components included deographic location was coded as a factor, environmental
similarity from euclidean distance of four edaphic variables including total
nitrogen, total carbon, moisture content, and bulk density, and
community similarity calculated from Bray-Curtis dissimilarity for abundance
data and the Jaccard index for presence/absance data. We tested the significance
of these components using a linear score test by comparing nested models with and
without each variance component \citep{qu2013}. We then estimated
the percent of variation in ecosystem function explained by each component 
including standard errors on that estimate 
using the \texttt{h2G} function from the
\texttt{gap} package. We then tested whether microbial community composition explains a
significant portion of the variation in ecosystem function after accounting for
variation in the environment and accounting for the spatial distance between
samples. We did this by fitting an intercept-only model with variance components
for environmental similarity from euclidean distance of a matrix of
environmental covariates and spatial similarity
also using euclidean distance from the latitude and longitude coordinates where
the samples were collected. We then used a linear score test to compare
nested models with and without the microbial community variance component to
determine whether community composition adds any variance explained after
accounting for spatial and environmental covariates. We then estimated the
variance explained by each component (environment, geography, and community
composition) with standard errors of the estimates using the \texttt{h2GE}
function. 

After determining which variance components were significant, we tested the
effect of each ASV on the rate of ecosystem function. We fit a variance
component model for each ASV with ASV abundance as the fixed effect and random
effects for each of the variance components. This tests the relationship between
taxon abundance and the rate of ecosystem function after controlling for
population stratification and data structure. The community similarity matrix
was reconstructed for each model to exclude the predictor ASV. 
Significant ASVs were identified by controlling the false discovery rate using
the \texttt{qvalue} package.

\section{Results}


\includegraphics[width=0.5\textwidth]{lowk_pmoa}
\includegraphics[width=0.5\textwidth]{lowk_pc1}
\includegraphics[width=0.5\textwidth]{lowk_rich}
\includegraphics[width=0.5\textwidth]{lowk_shan}

All three variance components are significant individually. Community
significant in the community + environment model and geography
significant in the geography + environment model. Table \ref{tab:pvalues}

\subfile{../tables/var_comp_pvalues}

Including different covariates removes different numbers of
significant taxa. Some covariates add taxa that were not previously
identified as significant. Table \ref{tab:n_taxa} 

\subfile{../tables/all_taxa_n}

After controlling for geographic proximity, environmental similarity, and
similarity in community composition, four taxa were significantly correlated
with the rate of methane oxidation. None of these four taxa are known methane
oxidizers or methane producers. Table  \ref{tab:sig_taxa}

\subfile{../tables/all_taxa_sig}

\section{Discussion}

In this paper, we argue that the traditional approach to microbial BEF research
ignores important understandings about the nature of ecological processes in
determining the distribution of microorganisms. The marker gene hypothesis has
proven unsuccesful in predicting the rates of ecosystem functions from the
abundance of functional genes and in our example study was not significant.
Similarly, diversity metrics from the entire Prokaryotic community were not
explanatory. These results agree with recent reviews of this literature and
demonstrate that these approaches are not a fruitful avenue for elucidating
microbial structure-function connections. 

Instead, microbial ecologists should treat microbial communities as analogous to
organisms with the metagenome representing the community genome. In this case,
ecosystem function represents a phenotype at the community level. As in
organismal biology, this ecosystem phenotype will be a mix of genotypic effects and
environmental effects. Using this conceptual framework, population genetic
approaches can be applied to microbial community datasets to identify  links
between the composition of microbial communities and the rates of ecosystem
functions.

There are two ways to interpret an association between taxon abundance and rate
of function. One is a statistical co-occurrence and the other is a biological
interpetation, for example pH affects function and pH affects taxon or pH
affects taxon, which affects function.
A significant association between a taxon and the rate of ecosystem function
could be explained a number of ways. One possibility is that the taxon is
statistically related, because it is causally connected to the function. This
could be direct, for example an organism that consumes methane, or indirect, for
example an organism that regulates substrates necessary for consumers of
methane. In either of these cases, this would represent a taxon that could be
useful as a biomarker of function or an organism to investigate to understand
the mechanistic relationship between communities and function. Alternativelly, a significant
association could occur due to admixture within a community. In a community of
microorganisms, any organism that tends to be in high abundance under the same
conditions as methane oxidizers will be correlated with methane oxidation even
if it has no mechanistic relationship with the rate of methane flux. In fact,
any taxa in high abundance in highly methanotrophic communities will be
statistically related to function.

In this variance component modeling framework, many taxa that are significantly
associated with function become non-significant after accounting for different
covariates. This could be explained by different relationships between the
environment, community assembly processes, and the rate of ecosystem function.
For example, a taxon significantly correlated with function that becomes
non-significant after controlling for the underlying environmental variation
could be acting as a conduit through which the environment alters function.
Alternatively, if a taxon becomes non-significant after controlling for
geographic proximity, this taxon could be related to function due to historical
effects such as past environmental conditions or through dispersal from a nearby
patch. Alternatively, geography could represent unmeasured environmental
variables that change over distance. Finally, if incorporating community
similarity makes a taxon non-significant, then 

None of the significant taxa identified by the full variance component model 
fall into expected functional groups. For example, none of the taxa identified
in this analysis were methanotrophs or methanogens.

Broad vs narrow processes: Some genetic diseases such as Parkinson's 
are controlled by a single locus. However, in most GWAS studies the phenotype of
interest is a result of many loci of small effect (as well as environmental
factors) such as human height and LDL cholesterol. For ecosystem functions that
consist of physiologically or phylogenetically "narrow" process (sensu
\cite{schimel1995b}), it's possible that a single marker gene could control the
rate of a process. For example, methane flux in permafrost in Sweden may be
controlled by a single taxon \citep{mccalley2014}. However, most microbial ecosystem functions are likely the result
of many taxa interacting directly or indirectly with function.
Trying to identify many taxa of small effect is made easier by constraining that
variation by defining your population in a way tha makes them similar: 
similar ecosystem, soil type, and abiotic conditions.

With these data, we demonstrate the
importance of selecting the appropriate population to sample from. We are
precise about the variation we are measuring. Specifically, we measured the rate
of a specific microbial pathway. In order to minimize environmenal variation, we
brought intact soil cores into the lab to incubate them under similar
temperature and methane concentration. We also statistically controlled for
environmental covariates that we couldn't isolate in the lab including soil
carbon and nitrogen, soil bulk density, and soil moisture content. Finally, we
surveyed the microbial community agnostically for microbial markers of ecosystem
function.

While this analysis focuses on using genome-wide association studies as a model
for mapping microbial community structure to ecosystem-function, other designs
are possible. For example, artificial ecosystem selection is a viable approach
to enrich for communities with high or low rates of ecosystem function
\citep{swenson2000, panke-buisse2015}. While community-wide association or
ecosystem-wide association or metagenome-wide association or microbiome-wide
association studies have sufficient power to identify taxa that are individually
associated with ecosystem function, whole-community selection could identify
assemblages of organisms with high rates of function that often co-occur in
productive ecosystems.

Another possibility that has not to our knowledge been demonstrated with any
microbial datasets is to apply the ecological form of the Price equation to
estimate the effects of microbial biodiversity on ecosystem function
\citep{fox2006, bannar-martin2018}. Using this equation, a change in ecosystem function between two, matched
observations can be attributed to species additions, species losses, and
changes in abundance of shared species \citep{fox2006,
bannar-martin2018}. To
calculate these vectors, function must be attributed at the species level (e.g.
biomass measurements of individual plant species). To apply this method to
microbial communities we will need to measure ecosystem function on a
per-species basis. There are several ways to do that including quantitative stable-isotope
probing to measure the accumulation of a substrate in the DNA of a portion of
the microbial community \citep{hungate2015}, NanoSIMS to measure the incorporation of nutrients at
the single-cell level \citep{mayali2012}, and isotope source partitioning coupled with measurements
of microbial pathways through quantification of marker genes through qPCR or
metagenomics/metatranscriptomics.

\subsection{Caveats}

Future studies could improve on the approach illustrated here in a number of
ways. First, GWAS studies rely on high power to detect significant associations
between phenotype and SNPs of small effect. This study consistent of 44 samples,
which is typical for microbial ecology sudies, but would benefit from a large
sampling of ecosystems and variation in both community composition and function. 
Second, this exploratory study sampled across
a diversity of wetland and upland ecosystems in the Congo basin. However, upland
ecosystems such as forests are known to have the greatest rates of high-affinity
methane oxidation. Identifying connections between variation in community
composition and variation in ecosystem function would benefit from sampling a
population of ecosystems that is known to exhibit variation in the rate of
the target function. Third, it is important to define a narrow pathway
definition. Functions such as low-affinity methane oxidation and bulk methane
production are relatively broad pathways and as such exhibit no significant
correlation with individual taxa \citep{meyer2019}. Here we have measured
high-affinity methane oxidation, which is performed by relatively few taxa.
Functions such as this are more phylogenetically conserved and as such are more
likely to have significant connections between community and function. Fourth,
these environmental variables measured may or may not be the correct ones to
measure to accurately account for the environmental effect on function.
Moisture, structure, total nitrogen, and total carbon are relatively course
measurements of environmental variation. Including more environmental variables
and variables known to influence the rate of the function of interest would more
accurately estimate the effect of the environment of communities and function.
This could also be applied to the geographic location of samples. Our study
sites were clustered around three locations that are far apart and so didn't
fully capturing the geographic variation underlying these ecosystems. Sampling
in a grid or logarithmic pattern might more suitable capture variation across
the landscape. 

This is novel, because it represent the first time that anyone has treated an
ecosystem like an organism. Further, it is the first time that anyone has
used this GWAS-esque approach while accounting for both environmental and
spatial structure in the data. 

\section{Conclusions}
Microbial community ecosystem function research has demonstrated that
functional gene abundance is not a good predictor of ecosystem function and
that diversity metrics do not add much to ecosystem models. Therefore, we
argue that microbial ecologists should look more agnostically at microbial
community structure to identify whether microbial community composition is
important for determing the rate of ecosystem function across space, time,
and ecological conditions. In this way we can identify which taxa or aspects of
microbial community structure determine the rates of ecosystem functions and so
will be useful for predicting and managing those rates. 

The problem of
connecting a complex microbial community matrix with an ecosystem-scale function
is not unique to microbial ecology. Population geneticists attempting to like
organismal phenotypes to its genotypic and environmental basis have dealt with
this problem for decades. In order to make robust associations between the
presence or abundance of specific microbial taxa and the rate of ecosystem-scale
functions, we should take lessons from population genetics in identifying
correlations between community structure and ecosystem function. We suggest
three approaches to addressing this question:

\begin{enumerate}
  \item Metagenome-wide association studies/Community-wide association
    studies/Ecosystem-wide association studies to survey natural ecosystems
    for potential biomarkers or drivers of ecosystem function. Appropriately
    control for community structure patterns to make robust
    structure-function associations. Combine this
    with experimental manipulations such as reciprocal transplants or
    environmental treatments to identify community-environment (CxE)
    interactions.
  \item Use artificial ecosystem selection to enrich for groups of taxa that
    respond to selection on a function. Use this to identify organisms that
    might play a critical role in ecosystem function. Use a response to
    selection to calculate ``community heritability''or the prortion of
    variation in function due to community composition.
  \item Measure rates of ecosystem functions on the individual cell or
    individual taxon level in order to partition changes in ecosystem
    function to gains, losses, or changes in abundance of
    specific taxa using the Price equation. Accomplish this by combining
    isotope source partitioning with functional group assessments, using
    labeling approaches like stable-isotope probing, ...
\end{enumerate}

\bibliographystyle{ecology} 
\bibliography{library}

\end{document}
