\documentclass{article}
%\usepackage[utf8]{inputenc}
%\usepackage{amsmath}
%\usepackage{amsfonts}
%\usepackage{amssymb}
%\usepackage{textalpha}
\usepackage{natbib}

\bibpunct{(}{)}{,}{a}{}{;}

\title{Linking microbial communities to ecosystem functions: what we can learn
from genotype-phenotype mapping in organisms} \author{Andrew H. Morris, Brendan
J. M. Bohannan}

\begin{document}

\maketitle



Deadline: August 1st

Theme: Conceptual Challenges in Microbial Community Ecology

Journal: Philosophical Transactions B

Max length: 8-9 pages

Data must be submitted as supplementary information in a repository with
accession number, link, or DOI

Supplement: They have encouraged us to include additional information including
data, tables, figures, text, images, videos, or recordings.

\section*{Tasks} 
\begin{enumerate}
\item What is novel about this? The first anyone has treated an ecosystem like an
organism. The first time anyone has done this correlative approach well
by considering environmental and spatial structure in the data. 
\item Figure out variance partitioning (with Hannah)
\item Maybe variance partitioning plus correlations (varPart)
\item Do PC correction in all combinations
\item Identify significant taxa and build a venn diagram figure (or
something)
\item 
\end{enumerate}
\section{Introduction} 

I want to know whether microbial community composition is important for determining ecosystem-scale functions independent of the underlying environmental variation.

Microorganisms mediate many biogeochemical processes, but despite 
much effort to link microbial communities
to these functions the mechanism by which they regulate these
processes remains unknown. Ecologists have investigated the effects of changing
biodiversity on ecosystem function for decades, but many of their approaches
are not applicable in microbial ecology due to the nature of microbial data. 
This limits our ability to predict and manage globally relevant biogeochemical cycles. To
address this challenge, we propose integrating current theory on community
assembly and the functioning of ecosystems with lessons from genotype-phenotype
mapping in quantitative genetics. This will address the
question of how does variation in microbial community structure contribute to
variation in the rate of ecosystem function? 

\subsection{Biodiversity increases ecosystem function in macrobial communities}


Local species richness is correlated with ecosystem function
\citep{hooper2005}. Most BEF research has focused on terrestrial plant 
communities where species losses lead to decreases in community abundance or biomass
\citep{naeem2003}. There are
two major explanations for this phenomenon. The first is the sampling effect
where a more diverse community is more likely to have one particularly
productive species \citep{cardinale2006}. The second is niche complementarity where more species are
more likely to occupy separate niches within the same ecosystem resulting in
more efficient resource use and greater biomass.

Regional species diversity may also be important for
local ecosystem functioning when metacommunity dynamics regulate the
diversity and composition of communities \citep{leibold2017}. 
Typical BEF research focuses on local selection
as a driver of ecosystem function. However, if regional
diversity is important, then dispersal processes could alter local species diversity and composition resulting in
a change in ecosystem function. This model is referred to as Community Assembly
and the Functioning of Ecosystems (CAFE). Community assembly can alter ecosystem
function if limiting dispersal reduces the likelihood of a highly productive
species from dispersing to a local site or if high dispersal of an unproductive
species overwhelms local selection.

The CAFE approach has been demonstrated using the Price equation where a change in ecosystem function between two, matched
observations can be attributed to species additions, species losses, and
changes in abundance of shared species \citep{fox2006,
bannar-martin2018}. To
calculate these vectors, function must be attributed at the species level (e.g.
biomass measurements of individual plant species). The CAFE framework demonstrates
that not only local richness, 
but also the composition of communities is important to the functioning of ecosystems.

\subsection{Current approaches to microbial BEF reveal little about 
structure-function mapping}

The trajectory of microbial BEF research has differed from that of macrobial BEF
for three reasons. First, microbial communities are
immensely complex with orders of magnitude more taxa than macrobial communities
and great intra-taxon variation for certain traits. Second, the way we measure
microbes using marker gene sequences does not allow us to attribute portions of
a function of interest to individual taxonomic groups . We can only
measure function in aggregate across an entire community. Finally, we cannot be
sure that the taxonomic groups that putatively perform a function are the ones
contributing in this moment.

As a result of these limitations, microbial ecologists have taken a different
approach to assessing BEF relationships for microbial communities than others
have for macrobial communities. This is generally a correlative approach where
certain community attributes that can be derived from a community matrix are
tested in terms of their relationship with function. These approaches make
different assumptions about how microbial communities should map onto function.
On one end of the spectrum, it is hypothesized that functional gene abundance - genes that
code for an enzyme that performs one step of a process - should be correlated
with the rate of the process associated with that gene. In this case, qPCR
is used to quantify the abundance of a functional gene to test whether it is
correlated with the rate of a process. This approach has had varied success and even the abundance of a
functional gene that putatively performs a process is rarely correlated with
that process \citep{rocca2015}. On the other end of the
spectrum, investigators make few assumptions about which members of the community should
alter function and so compute
taxonomic richness or other diversity metrics from the entire Prokaryotic
community inferred from 16S rRNA gene sequences. Taxonomic or functional
diversity of the Prokaryote community generally adds little variance explained
to models of ecosystem function \citep{graham2016}.

\subsection{Biology through analogy: structure-function mapping is similar
to genotype-phenotype mapping}

The challenges described above for mapping microbial community structure to
ecosystem function are analogous to the problem of identifying the
genetic basis of complex traits in organismal populations. In that case,
each individual organism has a level for a phenotype (e.g. presence
or absence of a disease state) as well as a vector of potentially
thousands of genetic loci with different alleles or single nucleotide
polymorphisms (SNPs). To identify the genetic basis for this trait, we sample
a population of organisms and try to correlate a matrix of SNPs with a
vector of phenotypes. In most cases, there are many more loci than
individuals, we cannot directly attribute a phenotypic state to a particular
SNP, and we do not know the function of genes associated with these
SNPs. This mirrors the problem of microbial structure-function mapping
and presents a potential set of analyses that could be applied in
microbial ecology.

In addition to the nature of the data, population geneticists have other
considerations when trying to make robust associations between phenotype
and SNPs. For example, loci within the genome that are close together tend
not to assort independently due to linkage disequilibrium. In addition,
organisms exhibit population stratification due to a variety of historical
and selective effects. For example, shared ancestry among European
populations of humans results in an association between genotype and
geography across Europe \citep{novembre2008}. This population structure
can lead to false positives when trying to correlate human genotypes
with genetic diseases or other phenotypes such as height (citation).

In general, organisms exhibit population differentiation due to spatial
and environmental patterns \citep{wright1943}. For geography, local genetic
drift and reduced gene flow due to geographic separation can lead to population
differentiation \citep{wright1943}. In addition, environmental distance can lead to both selection
against migrants \citep{hendry2004} and local selection on endemic populations to
generate variation. While
these processes have primarily been described for macrobial populations, the
same principles apply to microorganisms and should be considered
in microbial structure-function mapping \citep{martiny2006}. 

\subsection{Reframing the question: Is variation in microbial community
structure associated with differences in ecosystem function across space,
time, and environments?}

Typical microbial ecosystem function research focuses on whether microbial
community composition is important at the ecosystem level
\citep{schimel1998, cavigelli2000, balser2005, graham2016}. Given that
microbial functional groups typically mediate biogeochemical processes
and so the loss of a functional group will certainly alter function,
the answer is "yes". However, this question does not address whether
microbial community variation is actually useful for predicting or 
managing ecosystem functions. To address that problem, we need to know
whether variation within a functional group matters to variation in
ecosystem function. If local selection always optimizes the available
biodiversity to maximize process rates given the underlying environmental
variation, then composition should not matter, because we should be able
to predict ecosystem function from abiotic factors alone. Environmental
treatments, common gardens, and reciprocal transplants have shown
different functional potential for different community compositions.
In addition, spatial patterns for microbial communities suggest that
there should be observable diversification due to environmental and
spatial processes \citep{wright1943, martiny2006}. Therefore, variation
in microbial community composition seems to result in variation in 
ecosystem function. However, in order for microbial community data to be
useful for ecosystem modeling, the challenge is to identify specifically what variation in community composition
is associated with function across space, time, and environmental gradients.
To accomplish that, we need to separate the close association between environmental variation, community
composition, and ecosystem function to identify the aspects of microbial
biodiversity that are important for ecosystem function.

\subsection*{Extra text for 1.3/1.4}

\begin{enumerate}
    \item Things exhibit population differentiation due to spatial and environmental 
    patterns \citep{wright1943}
    \item Microbes exhibit similar patterns \citep{martiny2006}
    \item Pop gen tries to identify regions of the genome correlated with
    traits
    \item Considering ecosystems as individuals within a population presents
    an analogous problem of determining community markers of ecosystem traits (functions)
    \item Thus the goal of microbial BEF is to identify generalizable patterns
    of genetic variation in microbes to feed into hypothesis testing about
    potential mechanisms or to incorporate into ecosystem models
\end{enumerate}

Typically, microbial community structure-function research questions are framed
as does microbial composition matter to ecosystem function? The answer to this
question is obviously "yes" given that microorganisms mediate many of the
processes of interest. The question then becomes does microbial community
variation within a functional group or overall matter to ecosystem function? 
Generally it is assumed that microbial composition is functionally irrelevant, 
though studies of environmental treatments, common gardens, and reciprocal
transplants have shown that different community compositions are correlated with
different functions. The challenge with these studies is separating the close
connection between environmental variation, community composition, and ecosystem
function. Microorganisms cannot be useful to ecosystem modeling if there
is no identifiable variation in composition that consistently alters function
across space, time, and environmental gradients. The question "do we need to
include microorganisms in ecosystem models?" depends on identifying variation in
the community associated with function independent of the environment. If that's
true, then the next step in understanding or modeling these processes is 
identify what components of the community alter ecosystem function.

There are a number of ways that environmental or spatial covariates could alter
or obscure connections between microbial community structure and ecoystem
function. For example, microbial communities exhibit strong biogeographic
patterns \citep{martiny2006}. The historical effects of geographic proximity,
such as dispersal limitation, could lead to spurious associations between taxa
and function if taxa are unevenly distributed across the functional landscape.
Environmental selection can also shape microbial community structure, so
selection on the microbial community that is independent of the
process could lead to erroneous conclusions about the
underlying drivers of ecosystem function \citep{thompson2017}. For
example, pH is known to regulate both microbial biodiversity patterns
\citep{fierer2006} and methane cycling dynamics \citep{ye2012}, which could lead
to microbial taxa correlated with methane cycling, but mechanistically
unrelated. Environmental factors that directly influence function, such as
moisture, could alter function with or without affecting the community. 
Finally, community stratification where taxa co-occur in certain ecosystems due to
selection or historical effects for reasons unrelated to the ecosystem function
could lead to spurious associations.

\subsection{A new framework for structure-function mapping in microbial
communities}

Microbial ecosystem function could be assessed using statistical techniques
borrowed from quantitative genetics along with the conceptual framework for
community assembly processes provided by metacommunity theory and CAFE. This
would be a two-step process to first identify the portion of variation in
ecosystem function attributable to different aspects of an ecosystem: the
community, the underlying environmental variation, and to spatial processes.

\begin{equation}
V_F=V_E+V_S+V_C+\epsilon
\end{equation}

Ecosystem function is represented by $V_F$, $V_E$ is variation in the
environment and covariation between the community and the environment, $V_S$ is
variation in spatial distance between samples and covariation between the
community and space, and $V_C$ is variation in the community independent of
local selection and historical effects. The parameter $V_C$ also incorporates
unmeasured environmental and spatial effects. 

Each of these variance components also correspond to alternative hypotheses for
what drives ecosystem function. If most of the variation in $V_F$ is
attributable to $V_E$, then the environment drives function either directly
(abiotic processes) or indirectly by selecting for a particular microbial
community that maximizes function. In this model, the community is simply a 
conduit through which the environment acts on function. Alternatively, if $V_S$
explains most of the variation in $V_F$ then this suggests that function is
primarily a result of historical contingency and differences in dispersal traits.
Finally, $V_C$ could explain most of the variation in $V_F$. This would be
consistent with reciprocal transplant and common garden experiments showing that
community composition drivers ecosystem function independent of the environment.

This framework allows us to address the following questions: 1. Does variation
in microbial community structure explain a measurable portion of the variation in
ecosystem function or can all of that variation be attributed to the underlying
effect of the environmental and spatial processes? If so, 2. Which members of
the community are significantly correlated with function after controlling for
the covariances of space and environment? Finally, 3. Do these taxa match our hypotheses about
what controls ecosystem function (e.g. functional group abundance) or are there
alternative mechanisms that need to be explored?




\section{Methods}

Sample 16S rRNA gene sequences, environmental covariates, geographic distance,
ecosystem function. Use the DADA2 pipeline to assign ASVs. 
phylogenetic and taxonomic assignment Null
Models \citep{chase2011, stegen2015}

Would it be valuable to use a null model (as in Chase or Stegen) where we
generate random communities based on the distribution across samples and
abundance within samples of each taxon. Then try to correlate those abundances
with our observed rates of ecosystem function to see what distribution of
structure-function correlations to expect given our community data. These sort
of null community generating techniques can omit certain community assembly
processes (such as homogenizing selection or stochastic processes)

An alternative testing approach would be to use Hannah’s permutation test idea
where we randomly permute the process rates ~1000 times and rebuild the model
each time to see what the distribution of significant taxa is.

VC - Correlations between taxa and process determined by accounting for spatial
structure, environmental covariates, and community structure. Represents effect
of taxonomy, unmeasured environmental variables VE - Account for community
structure and correlate environmental variables cov(C, E) - The part of the venn
diagram that disappears when controlling for environment and space VC:VE - Test
the correlations between community and process rate at a variety of
environmental conditions.  epsilon/error term - Effects of natural variation,
unmeasured variables

Another modeling options is HMSC from \cite{ovaskainen2017} 

\section{Results}

For high-affinity methane oxidation in Gabon: We experimentally controlled some
environmental variables and our measured environmental variables are not
significantly correlated with process rate (C, N, WFPS) Space is correlated with
function



Be careful with attributing a certain \% of the variance to an individual taxa
when performing partial regression on the residuals. If a taxa explains 60\% of
the variance in the residuals after controlling for environment and space that
doesn’t mean it explains 60\% of the overall variance.

\section{Discussion}
Combination of qPCR and pathway partinioing, qSIP \citep{hungate2015},
and NanoSIMS \citep{mayali2012} may allow us to assign portions of
function to particular taxa or functional groups.
Problems that need to be addressed: We shouldn’t assume that functional groups
that putatively perform a process are the rate-limiting step or the strongest
correlate.  There may be unknown members of the community that perform or limit
a process The final step of the process may not be the rate-limiting step: it
may be substrate production or nutrient acquisition or something else (e.g.
Qusheng Jin’s work on modeling methanogenesis) We need to consider the
underlying biogeographic patterns shaping the distribution of microbial
communities (à la GWAS). 

\section{Conclusion}

\section*{Outline} 
\begin{enumerate} 
  \item Problem statement: Microorganisms putatively mediate biogeochemical
    cycles and yet we have been unable to demonstrate consistent BEF or CAFE
  relationships.   
\item  Opening: Biodiversity ecosystem function research has demonstrated the
  importance of protecting biodiversity to maintain ecosystem function.
  However, more recently the importance of community assembly processes such as
  dispersal have shown that regional diversity is potentially as important as
  local processes. Therefore, metacommunity dynamics should play a role in local
  ecosystem functioning.  
  \item Challenge: Applying CAFE to microbes would be
  convenient, however microbes pose several unique challenges. 1. Microbial
  communities are orders of magnitude more complex than macrobial communities.
  2. We don’t know who performs each function of interest. 3. We cannot
  attribute a portion of the rate of an ecosystem function to a particular
  individual or species like we can in plant communities.  
  \item Solution: The
  microbial community assembly - ecosystem function relationship is more
  analogous to the problem of mapping complex phenotypic traits to genetic
  variation in organisms. In this case we have thousands of genetic loci with
  unknown functions that may contribute positively or negatively (or not at all)
  to a particular trait.  
  \item Studies linking genotype to phenotype have to
  consider similar processes as metacommunities within the
  metapopulation such as confounding geography with genotypic similarity.
  Applying this to microbial metacommunities, associations between microbes and
  function could be due to a number of factors other than the “genetic” effect
  of microbial community structure on ecosystem function. By accounting for the
  covariance structure (space, environment, community interactions) we can
  identify true structure-function connections.  
  \item Approach: Demonstrate the
  previous approaches to microbial structure-function (e.g. abundance-function,
  diversity-function, PCA or indicator species analysis) and demonstrate our new
  approach, which accounts for metacommunity dynamics in local ecosystem
  function. Step 1. Variance partitioning between community, environment, and
  space. Step 2. GWAS-esque association mapping to identify microbes.  
  \item Take home: The way we study the functioning of microbial communities is
  inherently limiting in our ability to apply BEF or CAFE approaches. Instead,
  we need to learn from GWAS to identify robust structure-function
  relationships. What we learn from this is that 1. Contrary to past results,
  microbes do explain a substantial portion of the variation in ecosystem
  function when we account for other sources of variation. 2. Stochastic
  processes can alter the relationship between environment/community and
  function 3.  we cannot assume that we know what part of the community controls
  ecosystem function (e.g. functional genes) i.e. non-intuitive features can
  explain a lot of residual variance.  
  \item Conclusion: This work presents a
  framework for testing the relationship between microbial community structure
  and ecosystem function and demonstrates the importance of understanding
  microbial genetic variation in predicting ecosystem function.  
  \end{enumerate}

\section*{Possible things to include:}
Why do we think that communities will be important for understanding ecosystem
function?



\begin{equation}
V_F=V_E+V_C+V_EV_C+cov(C,E)+\epsilon
\end{equation}
\begin{equation}
F_{ij}=C_i+E_j+CE_{ij}+\epsilon_{ijk}
\end{equation}

In this equation, VF represents variation in the rate of an ecosystem function.
VE is variation in the environment that independently contributes to the rate of
the ecosystem function. VC is variation in microbial community structure (i.e.
composition, abundance, or diversity) that independently contributes to
ecosystem function. The interaction term VE:VC represents situations in which
the difference in function between two communities varies by environment. For
example, community A could outperform community B in environment 1, but not in
environment 2. Alternatively, community A could outperform B in environment 1
and environment 2, but the magnitude of the difference varies by environment.
The term covC,E captures covariance between community and environment where a
change in the environment produces an associated change in the community.
Finally,  is an error term for natural variation in these relationships. 

In the context of deterministic versus stochastic/neutral community assembly,
each portion of the variance represents different assembly processes. The cov(C,
E) represents selection whereas VC represents neutral processes of drift,
homogenizing dispersal, or dispersal-limitation. The interaction term VC:VE
represents differences in the rate constants/Q10s for different communities or
neutral processes of community assembly under a particular environmental
condition. VE is independent of the community and so doesn’t represent a
community assembly process, but is simply a physicochemical driver of VF such as
gas diffusion through water, tortuosity of soil pores, redox conditions, … I’m
not sure this paragraph is correct

VE = Direct physicochemical effects on process rates including abiotic
reactions, diffusivity, temperature limitation/Q10 (that doesn’t affect taxon
abundance or activity but does affect e.g. reaction rates), soil stabilization
of chemicals (sorption), tortuosity of soil pores, wind speed (affecting
differential pressure between soil air and atmosphere), concentration gradient
between soil and atmosphere.

cov(C, E) = Environmental selection on soil microorganisms that directly
influence function. Environmental variables that directly affect function and
also select for a microbe unrelated to function. An environmental variable that
both directly affects function and indirectly affects function via a microbe. 

VC:VE = Context-dependent organismal traits including different temperature
optima, substrate affinities, enzyme specificity, redox status/oxygen tolerance,
pH optima. 

Epsilon = Stochastic processes. Taxa occurrences or
environmental variables completely unrelated to function.

I can imagine several mechanisms by which a significant correlation between a
taxon and a process rate might be detected.  An environmental variable directly
affects function and selects for a microbe unrelated to function (false
positive). (VE) This would be removed by correcting for environment, but would
remain if it was due to an unmeasured environmental variable (and so would
appear in VC).  An environmental variable selects for a microbe that directly
affects function (cov(C, E)). This would be removed by correcting for the right
environmental variable, but would be revealed by controlling for everything
except the environment.  A taxon directly affects function (true positive)  and
co-occurs with another microbe unrelated to function (false positive) (This
could be VC or cov(C,E)). False positive would be removed by correcting for
community similarity. True positive would be lost if perfect covariance between
function, true positive, and false positive.  Function exhibits a spatial signal
(due to underlying spatial VE or VC) and a taxon exhibits a similar spatial
signal, but for different reasons (selection on different VE or VC or historical
effects (e.g. dispersal limitation)). Would be removed by correcting for space,
but would not be removed if there was a perfect covariance.

Why is it important to understand VC?  At this point in the development of
microbial ecology, we know that both selection and dispersal limitation shape
microbial community assembly. At the extremes, a single bacterial phylotype
cannot instantly be present in all environments. Likewise, oxygen will kill
obligate anaerobes. It also appears, to the best of our knowledge, that
microbial communities of different origins exhibit different functional rates
when introduced to identical environmental conditions. Therefore, environmental
selection on microbial communities cannot fully explain variation in ecosystem
functions if they are mediated by microorganisms. Put another way, even if we
could perfectly measure VE, we cannot fully understand VF.

The question that follows from this is: Does VC matter on the scales relevant to
humans? To answer that question, we have to understand the magnitude of 

Null models of community assembly
 
VC or VE:VC could also represent unmeasured environmental variables.

\bibliographystyle{ecology} 
\bibliography{library}

\end{document}
