---
output: pdf_document
---

Response to Reviewers

Dear Dr. Martiny and Reviewers,

Thank you so much for your thoughtful and helpful reviews.

We have revised the manuscript to reflect your comments and have provided
specific responses to your comments below.

>Referee: 1
>
>Comments to Author(s) The manuscript ``Linking microbial communities to
>ecosystem functions: what we can learn from genotype-phenotype mapping in
>organisms'' by Morris et al proposes that integrating traditional
>biodiversity-ecosystem function research with genotype-phenotype mapping
>approach and suggests this  could provide the new perspective microbial BEF
>studies.
>
>I believe paper has a value in terms of expanding our tools to validate data
>from microbial BEF -particularly, identifying potential key taxa.  I have two
>major concerns;
>
>1. The paper set out with the projection that current approaches of testing
>   microbial BEF are not good enough. I do not see evidence how this approach
>   is better than on any existing approach (which link microbial diversity,
>   community composition with ecosystem functions). The manuscript does provide
>   an additional tool to potential identity of key taxa which are involved in
>   specific functions. Introduction should reflect that.

The intention of this paper is not to show that previous approaches are bad or
that this approach is “better.” Instead, the goal of this paper is to lay out
how this question has been addressed in the past, present a new way of framing
the question, and then demonstrate one example of how that new framing might be
used. We have revised throughout the paper to try to reflect that goal more
accurately. In particular, we have toned down the critique of past BEF work and
made the introduction clearer about the fact that we have learned a lot from
BEF research and that this is a new way of thinking about the problem. 

>2. Paper has not done justice to recent progress (since 2015), we made in
>   demonstrating microbial BEF from field to global scales. Few examples
>   provided below 
>
>   Jing et al (2015). Nature Communications 
>
>   Delgado-Baquerizo et al., (2016) Nature Communication 
>
>   Delgado-Baquerizo et al., (2017) New Phytologist 
>
>   There are several more!

We have incorporated these references into the brief review of BEF research and
our review spans the literature from 2000 to 2018.

>Other comments
>1. L26-36. There have been number of recent papers published on this topic
>   which provide evidence of microbial BEF. Please see above.  
We have edited the text in this section to include these papers.
>2. L79-81. But there are other papers which provide support for microbial BEF. 
>3. L91-95. Here and several times later on- authors used one paper to suggest
>   that either there is no or extremely limited evidence of microbial BEF.
>   However, authors should  balance that there are multiple papers which
>   provided strong evidence in support of microbial BEF in field conditions
Here and elsewhere we have made it clear what aspect of the BEF literature
that we are critiquing and we have separated the critique from the review of
literature in support of microbial BEF.
4. L290-292. Can phage treatment be useful here?  Phage treatment is a possible
   approach and we have added it with a citation.
5. Reference list needs be carefully formatted to meet the criteria of the
   journalisms.

We have added the papers above and emphasized the support for microbial BEF,
particularly in the first paragraph of the Intro and in section 1.1. We have
also softened the language surrounding what we don’t know to more accurately
capture the state of the literature. We have also been more specific about what
questions remain open, particularly in section 1.2. We have updated the
citation format to match the Phil Trans B style.

Referee: 2

Comments to Author(s) Overview: For decades, ecologists have been developing
theoretical and empirical approaches for understanding the contribution of
species to ecosystem function. While advances have been made, relationships can
still be hard to detect, especially in microbial communities where there are
many taxa that are interacting and contributing in complex ways to different
functions. The premise of the Morris et al. manuscript is that ideas and
approaches borrowed from population genomics can be used to better understand
contributions of taxa to ecosystem function. The authors argue that that
biodiversity ecosystem functioning (BEF) relationships in microbial systems
have shown promise, but that they are often clouded by covariation between
environmental filters and ecosystem function as well as correction levels that
do not match the data. Specifically, methods from genotype-phenotype mapping
provide three insights, “agnostic” mapping, correction for multiple comparison,
and the ability to account for covariation between variables such as
environmental factors, geography, and community composition. The authors use an
example of methane oxidation in soil cores where previous analysis showed no
link between microbial community structure and function. After correcting for
multiple comparisons and accounting for covariation, their approach identified
six uniquely associated taxa. The authors conclude that genotype-phenotype
mapping can improve our ability to link microbial taxa with ecosystem functions
independent of environmental filters and other factors that covary with
community structure.

Major concerns:

-- The premise of using genotype-phenotype mapping to understand BEF is
intriguing. So, I was excited to read this manuscript. A major concern is that
I thought that authors spent too much space in the relatively short manuscript
focusing on the shortcomings of BEF research (sections in Abstracts and 1.1 –
1.3) relative to what I would have liked to have learned about
genotype-phenotype mapping (1.4). The latter needs to be better developed in
order for the manuscript to have its desired effect, which I assume is to help
researchers working on BEF to incorporate genotype-phenotype mapping approaches
to their research. I’m afraid that in the current version, the manuscript falls
short of providing the necessary background and justification.

We have shortened the review of the BEF literature and have tried to be more
clear and concise about what aspect of BEF we are critiquing, insofar as it
helps develop the mapping framework. We have also tried to unpack some of the
analogies that we use and provide examples of what they might like like in
microbial communities. See, for example, the new text in sections 2.1 and 2.2. 

-- I sort of like the section advocating for an “agnostic” approach given that
we often don’t know a priori which taxa are contributing directly or indirectly
to a function. However, I was more skeptical about attributing multiple
comparisons to a genotype-phenotype approach. For example, it is well known
that Bonferroni corrections are severe and alternatives have been in place for
decades.  For example, without correction for false discovery rate (FDR), the
author report that 460 taxa contributed to rates of methane oxidation. In my
experience, all FDR-correction is doing is removing variables (in this case,
taxa) that have p-values above some adjusted threshold that is determined by
the total number of variables in a model.

The 460 taxa were significant after FDR correction, not before. we have updated
the text to make this clearer. Indeed, FDR correction sets a threshold for
significance that is less conservative than Bonferroni correction, but
ultimately it is just a significance threshold. It is true that correcting for
multiple comparisons does not come from population genetics. We have removed
the idea of correcting for multiple comparisons from the three “take-home”
points and have focused our text on describing the other two points. We have
also tried to make it clear that the main take home lesson from
genotype-phenotype mapping is not any particular statistical approach, but
rather a general way of thinking about these statistical relationships.

-- The section on population stratification starting on line 152 is
interesting. This doesn’t seem to be something that microbial ecologists often
do. I wonder if it would be helpful to expand upon this some more and perhaps
even show more explicitly how such spurious correlations could contribute
inferences either using the case study (methane oxidation) or another cartoon
type example, perhaps as a separate Box.

We have provided some examples of spurious correlations (using the methane
oxidation example) and have expanded on the conceptual ideas behind
stratification.

-- The section on high affinity methane oxidation needs work. I have no problem
with the case study per se, but I think it needs to be reworked if it is mean
to serve as a general example of a pipeline for how one might use
genotype-phenotype mapping to explore BEF. The second and third paragraphs of
this section don’t particularly seem aligned with the objective of the
manuscript.

Much of this section has been reworked and shaped into a Results & Discussion
section to interpret each step in the process and to better illustrate how and
why to perform each part of the analysis.

-- In the end, as someone who has never done genotype-phenotype mapping (but as
someone who has thought about BEF relationships from a statistical perspective)
I didn’t really come away feeling like a I had a deeper understanding or
appreciation for how I might use the authors’ recommended approach, but I think
with some more effort and revision, this could be achieved. Since this seems
like a fairly novel idea and one that spans disciplines, I would encourage the
authors to think about how the paper might serve in part as a primer that would
help readers understand the conceptual and statistical analogies between
genotype-phenotype mapping and BEF research.

This has been addressed by many of the other comments, particularly in
expanding on the analogy and how some of the conceptual components might apply
to microbial communities. We have also tried to make the purpose of the example
clearer and interpret the results more thoroughly. Finally, we have added a
caveats/future directions section that provides some suggestions of what other
changes could be made to improve on this approach in future studies. 

-- Thought based on and around text on lines 104. Examples of phenotypes like
height are classic example of a quantitative trait that is influenced by many
genes. Could statistical approaches from quantitative genetics be used in a BEF
context?

The way we think about this is that genotype-phenotype mapping is the purview
of population genetics. Quantitative genetics is simply a subset of population
genenetics that deals with continuous rather than discrete traits. Ecosystem
function is more in line with quantitative genetics, because we’re dealing with
quantitative traits of ecosystems (e.g, rates of fluxes of molecules), but
ultimately the statistical approaches are the same. We have used “population
genetics” throughout in order to consistently use one term and so the ideas
apply more generally.

Minor Comments

-- Line 40: Environmental DNA is introduced here without a description of the
type of sequencing data being discussed, which may be critical to understanding
the information gathered from the DNA, be it phylogenetic or functional-gene
markers. Later, in line 208, 16S sequences are discussed as a way to identify
organisms that does not provide functional information. Even later, in line
302, it is clarified that functional group abundance can be measured via
genetic markers from environmental DNA.

We revised line 40 to specifically describe the type of sequencing data (16S
sequences or metagenomic sequences or qPCR of functional genes). We revised
line 302 to specifically refer to qPCR of functional genes. In all cases, the
phrase “environmental DNA” has been replaced with the specific type of
sequencing data being discussed. Sections 1.2 and 2.1 expand on the idea behind
functional genes.

-- Lines 140-167: Consider splitting Section 1.6 into two sections, one on
multiple comparisons and one on covariation, to stay consistent with the
expected discussion of three insights as identified in the abstract.

We removed the multiple comparisons section as per above comments and created a
separate section regarding covariation.

-- Lines 150-151: Consider expanding upon commonly used corrections in GWAS
studies. An additional sentence of two about a couple common corrections may
make this section stand on its own and allow the methods to be clearer to the
reader.

We added a few sentences elaborating on PC correction and variance component
modeling. Also, we added text regarding the design of the experiment around a
homogeneous population, which is considered a good practice in GWAS.

-- Lines 203-204: An explanation of the importance of site ID and ecosystem
type in structuring communities would aid the reader in understanding what
Figure 2 adds to the example. Is Figure 2 included to show the reader that the
environmental variation, community structure and geographic distance structure
communities? In general, it wasn’t clear to me that Figures 1 and 2 were
critical or necessary to include.

We added more interpretation of Figure 2.

-- Lines 264-267: The proposed method for identifying taxa of small effect
could benefit from a clearer set-up. The idea is overshadowed by the discussion
of detecting loci of small effect in genotype-phenotype studies and needs a
stronger connection to microbial biodiversity ecosystem function relationships
to stand on its own.

We moved this up to section 2 and made it clearer how this would work in BEF
studies.

-- Lines 256-259: The idea of “uniquely associated” taxa should be introduced
earlier in the text, perhaps section 1.4. Without explaining the change in
question from associated to uniquely associated, the example loses some of its
power and the importance of genotype-phenotype mapping methods is undersold.

We added discussion of uniquely associated to the section on controlling for
population structure.

-- Lines 50-52: consider revising for clarity

This text has been rewritten and moved to the “Population Stratification”
section

-- Line 57. Typo at end of topic sentence.

The comma has been corrected to a period.

-- Line 262-263: Reword sentence for clarity

This text has been revised.

-- Line 235: Other statistical approaches used for understanding causal
relationships include structural equation modeling (SEM) and path analysis.
These techniques also provide opportunity to identify latent variables, i.e.,
variables that are likely important but are difficult to measure. Might such
approaches be mentioned, and could they be used in conjunction with findings
from genotype-phenotype mapping? For example, it would be impossible to build
an SEM with 406 taxa and make any sense of what was going on, but with a
reduced set of taxa (e.g., 6), a SEM becomes very tractable.

We have added some discussion of how to use the identified taxa including a
form of microbial polygenic risk scores and including the taxa in a structural
equation model (Section 3.3).

-- To what degree to microbial BEF studies suffer from the fact that we are all
using relative abundance data? This is an obstacle that genotype-phenotype
mapping doesn’t have to deal with, correct?

We added text describing this to the discussion.

-- No hyphen needed in “microbially-mediated”

We have changed all instances of “microbially-mediated” to “microbially
mediated.”
