Dear Brendan,

Your article RSTB-2019-0244 entitled "Linking microbial communities to ecosystem functions: what we can learn from genotype-phenotype mapping in organisms" for publication in the upcoming theme issue of Philosophical Transactions B has now been refereed.

Please see the referee reports at the bottom of this email. The reviewers have recommended major revisions which we hope that you will be able to make within 6 weeks. On resubmission the paper may be subject to further review and acceptance cannot be guaranteed at this stage.

Please especially consider Referee #2's suggestions, which are quite thoughtful and seem as though they have the potential to greatly improve the impact of the manuscript.

To revise your manuscript, log into https://mc.manuscriptcentral.com/issue-ptrsb and enter your Author Center, where you will find your manuscript title listed under "Manuscripts with Decisions." Under "Actions," click on "Create a Revision." Your manuscript number has been appended to denote a revision. You will be unable to make your revisions on the originally submitted version of the manuscript. Once the revised manuscript is prepared, you can upload it and submit it through your Author Center.

When submitting your revised manuscript, you will be able to respond to the comments made by the reviewer(s) in the space provided. You can use this space to document any changes you make to the original manuscript. In order to expedite the processing of the revised manuscript, please be as specific as possible in your response to the reviewer(s). IMPORTANT: Your original files are available to you when you upload your revised manuscript. Please delete any redundant files before completing the submission.

We will need:
1) Text file of the manuscript (Word document, tex or similar) including an abstract/summary (up to 200 words), 3-6 keywords, references, tables (including captions), and figure captions. Please visit https://royalsociety.org/journals/authors/author-guidelines/ for full details.

2) Individual high resolution electronic files for each figure in the article (EPS or print-quality PDF preferred (either format should be produced directly from original creation package), or original software format).

3) All supplementary materials accompanying an accepted article will be treated as in their final form. Note that the Royal Society will not edit or typeset supplementary material and it will be hosted as provided. Please ensure that the supplementary material includes the paper details where possible (authors, article title, journal name). Supplementary files will be published alongside the paper on the journal website and posted on the online figshare repository (https://figshare.com). The heading and legend provided for each supplementary file during the submission process will be used to create the figshare page, so please ensure these are accurate and informative so that your files can be found in searches. Files on figshare will be made available approximately one week before the accompanying article so that the supplementary material can be attributed a unique DOI.

Again, thank you for contributing to this themed issue of Philosophical Transactions B and I look forward to receiving your revision.

Yours sincerely,

Dr Jennifer Martiny
jmartiny@uci.edu

Referee reports:

Referee: 1

Comments to Author(s)
The manuscript " Linking microbial communities to ecosystem functions: what we can learn from genotype-phenotype mapping in organisms' by Morris et al proposes that
integrating traditional biodiversity-ecosystem function research with genotype-phenotype
mapping approach and suggests this  could provide the new perspective microbial BEF studies.

I believe paper has a value in terms of expanding our tools to validate data from microbial BEF -particularly, identifying potential key taxa.  I have two major concerns;

1. The paper set out with the projection that current approaches of testing microbial BEF are not good enough. I do not see evidence how this approach is better than on any existing approach (which link microbial diversity, community composition with ecosystem functions). The manuscript does provide an additional tool to potential identity of key taxa which are involved in specific functions. Introduction should reflect that.

2. Paper has not done justice to recent progress (since 2015), we made in demonstrating microbial BEF from field to global scales. Few examples provided below
Jing et al (2015). Nature Communications
Delgado-Baquerizo et al., (2016) Nature Communication
Delgado-Baquerizo et al., (2017) New Phytologist
There are several more!

Other comments
1. L26-36. There have been number of recent papers published  on this topic which provide evidence of microbial BEF. Please see above.
2. L79-81. But there are other papers which provide support for microbial BEF.
3. L91-95. Here and several times later on- authors used one paper to suggest that either there is no or extremely limited evidence of microbial BEF. However, authors should  balance that there are multiple papers which provided strong evidence in support of microbial BEF in field conditions
4. L290-292. Can phage treatment be useful here?
5. Reference list needs be carefully formatted to meet the criteria of the journalisms.

Referee: 2

Comments to Author(s)
Overview: For decades, ecologists have been developing theoretical and empirical approaches for understanding the contribution of species to ecosystem function. While advances have been made, relationships can still be hard to detect, especially in microbial communities where there are many taxa that are interacting and contributing in complex ways to different functions. The premise of the Morris et al. manuscript is that ideas and approaches borrowed from population genomics can be used to better understand contributions of taxa to ecosystem function. The authors argue that that biodiversity ecosystem functioning (BEF) relationships in microbial systems have shown promise, but that they are often clouded by covariation between environmental filters and ecosystem function as well as correction levels that do not match the data. Specifically, methods from genotype-phenotype mapping provide three insights, “agnostic” mapping, correction for multiple comparison, and the ability to account for covariation between variables such as environmental factors, geography, and community composition. The authors use an example of methane oxidation in soil cores where previous analysis showed no link between microbial community structure and function. After correcting for multiple comparisons and accounting for covariation, their approach identified six uniquely associated taxa. The authors conclude that genotype-phenotype mapping can improve our ability to link microbial taxa with ecosystem functions independent of environmental filters and other factors that covary with community structure.

Major concerns:

-- The premise of using genotype-phenotype mapping to understand BEF is intriguing. So, I was excited to read this manuscript. A major concern is that I thought that authors spent too much space in the relatively short manuscript focusing on the shortcomings of BEF research (sections in Abstracts and 1.1 – 1.3) relative to what I would have liked to have learned about genotype-phenotype mapping (1.4). The latter needs to be better developed in order for the manuscript to have its desired effect, which I assume is to help researchers working on BEF to incorporate genotype-phenotype mapping approaches to their research. I’m afraid that in the current version, the manuscript falls short of providing the necessary background and justification.

-- I sort of like the section advocating for an “agnostic” approach given that we often don’t know a priori which taxa are contributing directly or indirectly to a function. However, I was more skeptical about attributing multiple comparisons to a genotype-phenotype approach. For example, it is well known that Bonferroni corrections are severe and alternatives have been in place for decades.  For example, without correction for false discovery rate (FDR), the author report that 460 taxa contributed to rates of methane oxidation. In my experience, all FDR-correction is doing is removing variables (in this case, taxa) that have p-values above some adjusted threshold that is determined by the total number of variables in a model.

-- The section on population stratification starting on line 152 is interesting. This doesn’t seem to be something that microbial ecologists often do. I wonder if it would be helpful to expand upon this some more and perhaps even show more explicitly how such spurious correlations could contribute inferences either using the case study (methane oxidation) or another cartoon type example, perhaps as a separate Box.
-- The section on high affinity methane oxidation needs work. I have no problem with the case study per se, but I think it needs to be reworked if it is mean to serve as a general example of a pipeline for how one might use genotype-phenotype mapping to explore BEF. The second and third paragraphs of this section don’t particularly seem aligned with the objective of the manuscript.

-- In the end, as someone who has never done genotype-phenotype mapping (but as someone who has thought about BEF relationships from a statistical perspective) I didn’t really come away feeling like a I had a deeper understanding or appreciation for how I might use the authors’ recommended approach, but I think with some more effort and revision, this could be achieved. Since this seems like a fairly novel idea and one that spans disciplines, I would encourage the authors to think about how the paper might serve in part as a primer that would help readers understand the conceptual and statistical analogies between genotype-phenotype mapping and BEF research.

-- Thought based on and around text on lines 104. Examples of phenotypes like height are classic example of a quantitative trait that is influenced by many genes. Could statistical approaches from quantitative genetics be used in a BEF context?

Minor Comments

-- Line 40: Environmental DNA is introduced here without a description of the type of sequencing data being discussed, which may be critical to understanding the information gathered from the DNA, be it phylogenetic or functional-gene markers. Later, in line 208, 16S sequences are discussed as a way to identify organisms that does not provide functional information. Even later, in line 302, it is clarified that functional group abundance can be measured via genetic markers from environmental DNA.

-- Lines 140-167: Consider splitting Section 1.6 into two sections, one on multiple comparisons and one on covariation, to stay consistent with the expected discussion of three insights as identified in the abstract.

-- Lines 150-151: Consider expanding upon commonly used corrections in GWAS studies. An additional sentence of two about a couple common corrections may make this section stand on its own and allow the methods to be clearer to the reader.

-- Lines 203-204: An explanation of the importance of site ID and ecosystem type in structuring communities would aid the reader in understanding what Figure 2 adds to the example. Is Figure 2 included to show the reader that the environmental variation, community structure and geographic distance structure communities? In general, it wasn’t clear to me that Figures 1 and 2 were critical or necessary to include.

-- Lines 264-267: The proposed method for identifying taxa of small effect could benefit from a clearer set-up. The idea is overshadowed by the discussion of detecting loci of small effect in genotype-phenotype studies and needs a stronger connection to microbial biodiversity ecosystem function relationships to stand on its own.

-- Lines 256-259: The idea of “uniquely associated” taxa should be introduced earlier in the text, perhaps section 1.4. Without explaining the change in question from associated to uniquely associated, the example loses some of its power and the importance of genotype-phenotype mapping methods is undersold.

-- Lines 50-52: consider revising for clarity

-- Line 57. Typo at end of topic sentence.

-- Line 262-263: Reword sentence for clarity

-- Line 235: Other statistical approaches used for understanding causal relationships include structural equation modeling (SEM) and path analysis. These techniques also provide opportunity to identify latent variables, i.e., variables that are likely important but are difficult to measure. Might such approaches be mentioned, and could they be used in conjunction with findings from genotype-phenotype mapping? For example, it would be impossible to build an SEM with 406 taxa and make any sense of what was going on, but with a reduced set of taxa (e.g., 6), a SEM becomes very tractable.

-- To what degree to microbial BEF studies suffer from the fact that we are all using relative abundance data? This is an obstacle that genotype-phenotype mapping doesn’t have to deal with, correct?

-- No hyphen needed in “microbially-mediated”
