\documentclass{beamer}
\usepackage[version=3]{mhchem}
\newcommand\blfootnote[1]{%
  \begingroup
  \renewcommand\thefootnote{}\footnote{#1}%
  \addtocounter{footnote}{-1}%
  \endgroup
}
\graphicspath {{../Figures/}}
\title{Linking microbial communities to ecosystem functions: }
\subtitle{What can we learn
from genotype-phenotype mapping in organisms?}
\author{Andrew Morris and Brendan Bohannan}
\date{May 9, 2019}
\institute{Philosophical Transactions B: Conceptual Challenges in Microbial Community Ecology}
\begin{document}


\frame{
\titlepage
}

\frame{
  \frametitle{My goals for today}
  \begin{enumerate}
    \item Present my latest ideas for this conceptual framework
    \item Get feedback on the logic/conceptual ideas. 
      \begin{itemize}
        \item cool things
        \item plot holes
      \end{itemize}
    \item Get your ideas on what analyses to include
  \end{enumerate}
}

\frame{
  \frametitle{Problem Statement}
  \centering
  \begin{itemize}
    \item Microorganisms putatively mediate biogeochemical cycles
    \item However, we have not yet demonstrated microbial biodiversity-ecosystem
      function relationships\footnotemark
    \item This limits our ability to understand and model these processes
  \end{itemize}
\includegraphics[width=0.8\textwidth]{cycle_crobes.pdf}
\footnotetext{Rocca et al. 2015, Graham et al. 2016}
}
\frame{
  \frametitle{Questions}
  \begin{enumerate}
    \item Do microbial communities explain a unique portion of the variance in
      ecosystem function (independent of environmental and spatial
      factors)?
    \item Which taxa are significantly correlated with function after controlling
      for space and environment?
    \item Do the identified taxa fit into expected functional groups?
  \end{enumerate}
}
\note{
Question 1 is if we control for the underlying envrionmental variables and for
the spatial proximity of sites, does the structure of microbial communities
alter ecosystem function? The reason this is important is if there is some
environmental variable that can reasonably predict function, it's a whole lot
easier to measure pH than to sequence microbial genes. So microbes will really
only be important for ecosystem models if there's something about microbial
physiology that is important for function independent of selection.

Assuming there is some variance there...2.

Question 3 is kind of a sub-question to 2. So, do methanogens explain most of
the variation in methane production or is it some other group e.g. substrate
producers?
}

\frame{
  \frametitle{Biodiversity-Ecosystem Function (BEF)}
\centering
Traditional BEF research has demonstrated for macrobial communities that greater
local richness can lead to greater overall ecosystem function. 
\includegraphics[width=0.8\textwidth]{hooper_biodiversity.png}\blfootnote{Hooper
et al. 2005}
}

\frame{
  \frametitle{Two explanations for BEF}
  \begin{itemize}
    \item Sampling/Selection Effect
    \item Niche Complementarity
  \end{itemize}
  \blfootnote{Hooper et al. 2005}
}

\frame{
  \frametitle{Community Assembly and the Functioning of Ecosystem (CAFE)}
  \centering
  More recently, the importance of regional biodiversity has been demonstrated
  in biodiversity experiments. In this case, when selection for a highly
  productive species is strong, there is a negative local richness-function
  relationship. \\~\\

  \includegraphics[width=0.6\textwidth]{cafe}\blfootnote{Leibold et al. 2017}
}
\note{
  Because the competitive dominant outcompetes other members of the community
}

\frame{
  \frametitle{Implication: Regional biodiversity can influence local ecosystem
    function via metacommunity dynamics. 
}
\pause
  \begin{itemize}
    \item e.g. At intermediate dispersal, when local selection is maximized, maximum ecosystem function is
      associated with a decrease in diversity (\textbf{species-sorting}). 
      \pause
    \item At lower dispersal rates, dispersal limitation weakens the selection
      effect and decreases
  function while increasing complementarity (\textbf{patch-dynamics}). 
  \pause
\item At higher dispersal rates, source-sink dynamics overwhelm local communities with less productive
  species resulting in an increase in diversity and a decrease in function
  (\textbf{mass-effects}).
\end{itemize}
\pause
\textbf{Therefore, spatial processes (metacommunity dynamics) could play an
  important role in local ecosystem function.}\blfootnote{Leibold et al. 2004,
Leibold et al. 2016}
}


\frame{
  \frametitle{Implementing CAFE: The Price Equation}
  \begin{itemize}
    \item The ecological version of the Price equation is used within the CAFE framework to partition ecosystem
  function into the effect of species gains, species losses, and changes in
  abundance (composition effects). 
\item This depends on assigning a portion of ecosystem function to individual species
(e.g. biomass of a single grass plant from a quadrat)
\end{itemize}
\textbf{CAFE demonstrates that not only is local richness important for
  functioning, but composition also plays a role in the functioning of
ecosystems}
  \blfootnote{Fox et al. 2006, Bannar-Martin et
  al. 2018}

}
\frame{
  \frametitle{How we measure microbes is different from macrobial BEF and CAFE}
  \pause
  \begin{itemize}
    \item We do not know who is performing the function at this time (e.g. dormancy,
    incomplete knowledge of functional groups)
    \pause
  \item We cannot directly attribute function to individual organisms or taxa with our
    current sampling methods (metabarcoding, metagenomics)
    \pause
\end{itemize}
\textbf{Therefore, we cannot (yet) use the Price equation for microbes}
}
\note{
The positive BEF relationships from the literature are looking at plant species
richness as it relates to ecosystem function NOT overall diversity. To put it in
microbial terms: They're correlating the diversity of photoautotrophs with the
rate of of photoautotrophy. We don't
know all of the taxa that necessarily contribute to function (or limit it
upstream).

A plant ecologist can pick up a single grass plant and say "This plant fixed thi
smuch carbon", but we cannot pick up a single methanotroph and say "This microbe
oxidized this many molecules of methane".
}

\frame{
  \frametitle{Microbial BEF so far}
  \framesubtitle{What \textit{have} we done?}
  \pause
  \begin{itemize}
    \item Assume functional group or functional gene abundance is correlated
      with the process performed by that group. \textbf{For most functions, that
      doesn't work}\footnotemark \\~\\
  \footnotetext{Rocca et al. 2015}
  \pause
    \item Explain variation in function with \textbf{functional diversity or
taxonomic diversity} \textit{of the entire prokaryotic community} inferred 
from 16S rRNA genes, which generally \textbf{achieves only a small amount of
explanatory power}\footnotemark.
\end{itemize}
  \footnotetext{Graham et al. 2016}
}

\frame{
  \frametitle{Microbial BEF so far, cont'd.}
  Generally ignore:
  \begin{itemize}
    \item Environmental covariates\footnote{except in Graham et al. 2016, but then microbes
      weren't very explanatory}
    \item Spatial proximity
    \item Community structure/co-occurrence patterns\footnote{One host-microbiome
      diabetes study does consider this (Qin et al. 2012)}
  \end{itemize}

}
\note{
  These studies generally do simple correlations between a single taxon or gene
  or diversity metric and function. 
}
\frame{
  \frametitle{Goal: Develop a framework that achieves the following}
  \begin{itemize}
    \item Use the CAFE framework to consider both effects of diversity and
      composition
    \item Incorporate spatial dynamics \textit{\'a la} Metacommunity Theory
    \item Do not assume anything about who is perfoming the function
\end{itemize}
}
\note{
  To achieve these goals, I actually think the BEF literature is not the right
  place to look
}

\frame{
  \frametitle{Biology through Analogy: microbial ecosystem function is less like
    BEF and more like genotype-phenotype mapping}

\begin{table}[ht]
\centering  
\begin{tabular}{rllllll}
  \hline
  Function & & Taxon & Taxon& Taxon& Taxon& Taxon \\ 
  \hline
  6.4 & & 0 & 10 & 5 & 3 & 0  \\
  12.1 & & 2 & 0 & 15 & 2 & 0  \\
  8.3 & & 3 & 0 & 17 & 1 & 0  \\
  0.4 & & 0 & 3 & 7 & 0 & 3  \\
  \\
  \hline
  Phenotype & & SNP & SNP & SNP & SNP & SNP \\ 
  \hline
  6.4 & & 0 & 1 & 1 & 1 & 0  \\
  12.1 & & 1 & 0 & 1 & 1 & 0  \\
  8.3 & & 1 & 0 & 1 & 1 & 0  \\
  0.4 & & 0 & 1 & 1 & 0 & 1  \\
   \hline       
\end{tabular} 
\end{table}

}
\note{

  The nature of microbial community data (community matrix of sequence variants
for thousands of taxa) and microbial function data (single measurement for an
entire ecosystem) makes this problem analogous to identifying the genetic
variation that underlies complex traits in organisms (i.e. a single phenotype
value per individual with thousands of SNPs).

In each of these cases the data are quite similar.
}
\frame{
  \frametitle{Genome-wide association studies}
  \begin{itemize}
    \item Robust correlations (adjusting for multiple comparisons)
    \item Correct for population structure
  \end{itemize}
  \centerline{
  \includegraphics[width=0.7\textwidth]{europe}
  \blfootnote{Novembre et al. 2008}
}
}
\frame{
  \frametitle{Underlying covariances could obscure structure-function
  relationships}

	\begin{columns}
		\begin{column}{0.3\textwidth}
			\centering
			\pause
			\includegraphics[width=\textwidth]{distance_decay.pdf}
		\end{column}
		\begin{column}{0.3\textwidth} \centering
			\pause
			\includegraphics[width=\textwidth]{community_environment.pdf}
		\end{column}
		\begin{column}{0.3\textwidth} \centering
			\pause
			\includegraphics[width=\textwidth]{similarity_of_function.pdf}
		\end{column}
	\end{columns}
}
\frame{
  \frametitle{Ways in which we could get false positives}
  \begin{enumerate}
    \item An environmental variable selects for a crobe that is correlated with
      function
    \item Sites closer together are more likely to have similar function and
      similar communities due to historical effects
    \item Co-occurrences among taxa due to species interactions. i.e. if taxon A
      is correlated with function and correlated with taxon B, then taxon B
      would be incorrectly correlated with function.
    
  \end{enumerate}
}

\frame{

	\frametitle{How does variation in microbial community structure contribute to
	variation in ecosystem function?}
	\centering
	\large
\pause	
  $V_F = V_E + V_C +$ 
  $V_EV_C + Cov(E, C) + \epsilon$ \\~\\

  \pause
	$V_F = V_E + V_C + V_S + \epsilon$
}
\note{
  I'm thinking about simplifying that equation by combining some of the terms
  and talking to Mathew Leibold helped influence me in this regard. I liked how
  he laid out his variance partitioning framework.  }

\note{
Okay, imagine a venn diagram with three circles. The overlap between Environment
and Space and covariance of environment and space. As per Liebold's suggestion,
incorporate 50\% of that overlap into each of E and S. The overlap between C-E
and C-S is environmental selection and spatial autocorrelation. Incorporates
those into E and S, respectively. The remainder of the third circle is variance
uniquely associated with variation in the community. The reponse variable is a
vector of ecosystem function (or a matrix if you consider multiple functions at
one)
}
\note{
  Variance in ecosystem function is a complex set of interactions between local
  environmental conditions, local community structure, spatial processes like
  dispersal, and stochastic processes. For example, the environment has direct
  effects on funtion via abiotic processes, but also influences function by
  selecting on the community. Spatial processes influence function through
  patterns like distance-decay where closer ecosystems tend to be more similar
  and organisms are more likely to disperse close-by. Stochastic processes like
  community drift and diversification influence the community. I'm choosing to
  simplify that variance partitioning problem by focusing on the three main
  sources of variation: effects of the environment (directly or indirectly via
  selection), effects of spatial proximity, and effects of co-occurrence
  patterns among taxa unrelated to function (community structure).
}


\frame{
  \begin{itemize}
    \item $V_F$ = Variation in ecosystem function
    \item $V_E$ = Direct effect of environment on function (abiotic) plus
      indirect effect via selection on the community
    \item $V_S$ = Effect of spatial proximity e.g. distance-decay
    \item $V_C$ = Community effect independent of the environment and spatial
      proximity, unmeasured environmental variables
    \item $\epsilon$ = error term, stochastic processes 
  \end{itemize}
}
\frame{
  \frametitle{How to do this?}
  \begin{itemize}
    \item varComp?
    \item HMSC
    \item ...?
  \end{itemize}
}

\frame{
  \frametitle{Which taxa are significantly correlated with function after controlling
  for space and environment?}
  My current approach:
  \begin{itemize}
    \item Partial regression on function and abundance to control for community
      similarity, environmental similarity, and spatial proximity
    \item Perform individual correlations between each taxon and function
    \item Correct for multiple comparisons using Bonferroni (alpha/n)
  \end{itemize}
}
\frame{
  \frametitle{Things I don't like about my current approach}
  \begin{itemize}
    \item Have to choose an arbitrary number of PC axes for partial regression
    \item Cannot estimate the variance components for Ve, Vc, and Vs
    \item People say it's out-of-date and it would be worth using a more modern
      mixed modeling approach
  \end{itemize}
  What should I do instead?
  \begin{itemize}
    \item Mixed modeling
    \item Matrix regression
  \end{itemize}
}
\frame{
  \frametitle{Possible Research Plan}
  \begin{enumerate}
    \item Variance partitioning between community, environment,
      space, and error.
    \item Identify significant taxa using statistical techniques from GWAS
    \item Use one or two datasets: Gabon and Brazil; One or more functions:
      High-affinity methane oxidation, low-affinity, methane production
    \item Compare results to other analyses that people typically do: indicator
      species analysis, PC regression...
  \end{enumerate}
}
\frame{
  \frametitle{Results}
  \begin{enumerate}
        \item \textasciitilde80 taxa correlated with function, decreases to \textasciitilde50 if I
          control for environmental/spatial covariates, decreases to \textasciitilde3 if I
          control for community structure
        \item Number of taxa varies dramatically depending on the number of PC
          axes I include
    \item This identified three taxa, but we don't know anything about their
      metabolism:
      \begin{itemize}
        \item Group 2 Acidobacterium
        \item Conexibacter (Actinobacteria)
        \item Armatimonas (Group 1 Armatimonadetes)
      \end{itemize}
    \item No methanotrophs!
  \end{enumerate}
}
\note{
  Possible explanations: Methane oxidizers are selected for under the
  environmental conditions and so act as conduits for the environment.
  Methanotrophs perfectly covary with some environmental variable or other
  community member or space. 
}
\end{document}
